main.py

```diff
@@ -1,25 +1,88 @@
 import numpy as np
 import pandas as pd
-from src import csv_read
-from src import DecisionTree
+from src import tree_read,neighbor_read
+from src import DecisionTree,NeighborsClassifier
 from matplotlib import pyplot as plt
 from sklearn import tree
 from sklearn.model_selection import GridSearchCV
+import os
+from datetime import datetime
 if __name__ == '__main__':
```

Comment:
Please put machine learning pipelines (regression, classification, clustering) into separate functions to avoid cluttering the main function